LoopBack/LBFile.m

```diff
@@ -85,34 +98,151 @@ - (SLRESTContract *)contract {
     [contract addItem:[SLRESTContractItem itemWithPattern:[NSString stringWithFormat:@"/%@/:container/files/:name", self.className]
+                                                     verb:@"DELETE"]
+            forMethod:[NSString stringWithFormat:@"%@.prototype.delete", self.className]];
+    [contract addItem:[SLRESTContractItem itemWithPattern:[NSString stringWithFormat:@"/%@/:container/upload", self.className]
+                                                     verb:@"POST"
+                                                multipart:YES]
+            forMethod:[NSString stringWithFormat:@"%@.upload", self.className]];
+    [contract addItem:[SLRESTContractItem itemWithPattern:[NSString stringWithFormat:@"/%@/:container/download/:name", self.className]
+                                                     verb:@"GET"]
+            forMethod:[NSString stringWithFormat:@"%@.download", self.className]];
+    [contract addItem:[SLRESTContractItem itemWithPattern:[NSString stringWithFormat:@"/%@/:container/files/:name", self.className]
                                                      verb:@"GET"]
             forMethod:[NSString stringWithFormat:@"%@.get", self.className]];
-    
+    [contract addItem:[SLRESTContractItem itemWithPattern:[NSString stringWithFormat:@"/%@/:container/files", self.className]
+                                                     verb:@"GET"]
+            forMethod:[NSString stringWithFormat:@"%@.getAll", self.className]];
+
     return contract;
 }
 
-- (LBFile *)createFileWithName:(NSString*)name
-                     localPath:(NSString*)localPath
-                     container:(NSString*)container {
-    LBFile *file = (LBFile*)[self modelWithDictionary:@{@"name" : name, @"localPath" : localPath, @"container" : container}];
+- (LBFile *)createFileWithName:(NSString *)name
+                     localPath:(NSString *)localPath
+                     container:(NSString *)container {
+    
+    LBFile *file = (LBFile *)[self modelWithDictionary:@{ @"name": name,
+                                                          @"localPath": localPath,
+                                                          @"container": container }];
     return file;
 }
 
-- (void)getFileWithName:(NSString*)name
-              localPath:(NSString*)localPath
-              container:(NSString*)container
+- (void)getFileWithName:(NSString *)name
+              localPath:(NSString *)localPath
+              container:(NSString *)container
                 success:(LBFileGetSuccessBlock)success
                 failure:(SLFailureBlock)failure {
+
     NSParameterAssert(name);
     NSParameterAssert(container);
     [self invokeStaticMethod:@"get"
-            parameters:@{@"name": name, @"localPath" : localPath, @"container" : container}
+            parameters:@{ @"name": name,
+                          @"container" : container }
                success:^(id value) {
```

Comment:
FYI, removal of localPath from parameters fixes the bogus string in the issued URL; i.e. 
Before `/containers/container1/files/uploadTest.txt?localPath=%2FUsers%2Fhideya%2FLibrary%2FDeveloper%2FCoreSimulator%2FDevices%2F2CDDA4C7-F0A4-4387-BAED-C0B116FBCB03%2Fdata%2Ftmp%2F`
After `/containers/container1/files/uploadTest.txt?`
It was actually bad since it exposes local file path information to the outside.